src/generic-container/started-generic-container.ts

```diff
@@ -28,23 +28,16 @@ export class StartedGenericContainer implements StartedTestContainer {
     return this.stopContainer(options);
   }
 
-  public async restart(options: Partial<RestartOptions> = {}): Promise<StartedGenericContainer> {
+  public async restart(options: Partial<RestartOptions> = {}): Promise<void> {
     const resolvedOptions: RestartOptions = { timeout: 0, ...options };
     await restartContainer(this.container, resolvedOptions);
-    const newInspectRes = await inspectContainer(this.container);
-    const newBoundPorts = await BoundPorts.fromInspectResult(newInspectRes).filter(
+
+    // Get the restart container with its new bound ports and update it
+    this.inspectResult = await inspectContainer(this.container);
+    this.boundPorts = await BoundPorts.fromInspectResult(this.inspectResult).filter(
```

Comment:
Do the ports change when the container is restarted?

Reply:
Yup. That's the reason why directly using `dockerrode` to perform a restart fails.

This is how I work around this limitation before:

```typescript
// Explicitly change the typing to any
const container: any = this.requireContainer();
const inspectRes = await inspectContainer(dockerrodeContainer);
const boundPorts: BoundPorts = (BoundPorts as any).fromInspectResult(inspectRes).filter(DeFiDContainer.DefiDPorts[this.network]);

// Since its a any type, typescript does not scream at us
// Javascript allows for this during run time
container.inspectResult = inspectRes;
container.boundPorts = boundPorts;
```